pg/wal/cursor_multi_insert_test.go

```diff
@@ -0,0 +1,39 @@
+package wal
+
+// Copyright 2015 MediaMath <http://www.mediamath.com>.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+import "testing"
+
+func TestCorrectCountOfMultiInsertRecords(t *testing.T) {
+	cptr, err := NewCursorAtPrevCheckpoint("./test_data/multi_insert")
+	if err != nil {
+		t.Fatalf("error creating cursor: %v", err)
+	}
+	c := *cptr
+
+	var (
+		expectedMultiInserts = 7313
+		actualMultiInserts   int
+	)
+
+	var ents []Entry
+
+	for {
+		ents, c, err = c.ReadEntries()
+		if err == nil && ents != nil {
```

Comment:
When `ents` is `nil` it signals that there are no new records to be read at this time.